src/context/AuthContext.tsx

```diff
@@ -0,0 +1,199 @@
+      method: 'eth_chainId',
+    })) as string;
+
+    // Wallet info setup in wallet state
+    setWallet({ accounts, balance, chainId });
+
+    // Initiate PrimeSDK after login withe metamask and setting up PrimeSDK in global state
+    const metamaskProvider = await MetaMaskWalletProvider.connect();
+    const prime_sdk = new PrimeSdk(metamaskProvider, { chainId: Number(chainId), projectKey: '' });
+    setPrimeSdk(prime_sdk);
+  }, []);
+
+  const updateWalletAndAccounts = useCallback(() => _updateWallet(), [_updateWallet]);
+  const updateWallet = useCallback((accounts: any) => _updateWallet(accounts), [_updateWallet]);
+
+  /**
+   * This logic checks if MetaMask is installed. If it is, some event handlers are set up
+   * to update the wallet state when MetaMask changes. The function returned by useEffect
+   * is used as a "cleanup": it removes the event handlers whenever the MetaMaskProvider
+   * is unmounted.
+   */
+  useEffect(() => {
+    const getProvider = async () => {
+      try {
+        const provider = await detectEthereumProvider({ silent: true });
+        setHasProvider(Boolean(provider));
+
+        if (provider) {
+          updateWalletAndAccounts();
+          window.ethereum?.on('accountsChanged', updateWallet);
+          window.ethereum?.on('chainChanged', updateWalletAndAccounts);
+        }
+      } catch (error) {
+        console.error('Error detecting Ethereum provider:', error);
+        setErrorMessage('Error detecting Ethereum provider');
+      }
+    };
+
+    getProvider();
+
+    // Cleanup function to remove event listeners
+    return () => {
+      try {
+        window.ethereum?.removeListener('accountsChanged', updateWallet);
+        window.ethereum?.removeListener('chainChanged', updateWalletAndAccounts);
+      } catch (error) {
+        console.error('Error removing event listeners:', error);
+      }
+    };
+  }, [updateWallet, updateWalletAndAccounts]);
+
+  // Function to connect MetaMask
+  const connectMetaMask = async () => {
+    setIsConnecting(true);
+
+    try {
+      // Request account
+      const accounts: any = await window.ethereum?.request({
+        method: 'eth_requestAccounts',
+      });
+      if (accounts.length > 0) {
+        // Request for sign verification
+        const message = 'Connect + Sign';
+        await window.ethereum?.request({
+          method: 'personal_sign',
+          params: [message, accounts[0]],
+        });
+        updateWallet(accounts);
+      } else {
+        console.log('account not found');
+      }
+    } catch (err: any) {
+      console.error('Error connecting MetaMask:', err);
+      setErrorMessage(err.message);
+    }
+    setIsConnecting(false);
+  };
+  // Function to log out
+  const logout = () => {
+    try {
+      sdk?.terminate();
+      setWallet(disconnectedState);
+      setIsConnecting(false);
+    } catch (error) {
+      console.error('Error logging out:', error);
+      setErrorMessage('Error logging out');
```

Comment:
```suggestion
      setErrorMessage('Error logging out, please try again');
```

Always remember that user facing messages should try to suggest what they should do next to help their problem

Reply:
current changes done and will keep in mind this comment